### PR TITLE
Fix PropTypes and component stack tests

### PR DESCRIPTION
**Pull request by marvinhagemeister**
These were broken again after merging #3002 (no worries, was expected to break)

### CODE REVIEW

File: debug/src/debug.js [hunk 1]
```diff
@@ -233,22 +233,23 @@ export function initDebug() {
 				}
 			}
 
-			let values = vnode.props;
-			if (vnode.type._forwarded) {
+			// If vnode is not present we're mounting
+			let values = vnode ? vnode.props : internal.props;
```
Comment (marvinhagemeister) [inline debug/src/debug.js:237]: FYI: At this point `internal` will point to the `props` object from the previous render, not the new one.

File: debug/src/debug.js [hunk 2]
```diff
@@ -162,67 +162,67 @@ export function initDebug() {
...
+					`by createRef(), but got [${typeof internal.ref}] instead\n` +
+					serializeVNode(internal) +
+					`\n\n${getOwnerStack(internal)}`
 			);
 		}
 
-		if (typeof vnode.type == 'string') {
-			for (const key in vnode.props) {
+		if (typeof internal.type == 'string') {
+			for (const key in internal.props) {
```
Comment (andrewiggins) [inline debug/src/debug.js:197]: Does it matter that this is looping over old props?

**Review by andrewiggins — APPROVED** [review]
Looking good! 🎉

File: debug/src/debug.js [hunk 3]
```diff
@@ -162,67 +162,67 @@ export function initDebug() {
...
 
 		if (
-			vnode.ref !== undefined &&
-			typeof vnode.ref != 'function' &&
-			typeof vnode.ref != 'object' &&
-			!('$$typeof' in vnode) // allow string refs when preact-compat is installed
+			internal.ref !== undefined &&
+			typeof internal.ref != 'function' &&
+			typeof internal.ref != 'object' &&
+			!('$$typeof' in internal) // allow string refs when preact-compat is installed
```
[the remark this answers is not in the document]
Comment (marvinhagemeister) [reply debug/src/debug.js]: Good catch! You're right on the money. It should be `vnode` here.

Realized that we do two kind of checks in `preact/debug`:

1. `vnode` / `internal` validation. Things like `props` etc
2. Hierarchical validation: Do I have the right parent, etc

The first one is much easier to do in `options._internal` and the second one in `options._diff` as it relies on the tree partially being there.

[the remark this answers is not in the document]
Comment (marvinhagemeister) [reply debug/src/debug.js]: Although,... wait no that doesn't work. We only create the internal once on mount😅

**Pull request merged**